source/fem-dem/ib_particles_dem.cc

```diff
@@ -655,113 +835,323 @@ IBParticlesDEM<dim>::calculate_pw_lubrication_force(
+               best_index_of_face != best_cells.end();
+               ++best_index_of_face)
+            {
+              auto &boundary_cell = best_index_of_face->second;
 
-          Tensor<1, 3> point_to_particle_vector =
-            particle_position_3d - point_on_boundary_3d;
+              auto         boundary_cell_information = boundary_cell;
+              Tensor<1, 3> normal =
+                tensor_nd_to_3d(boundary_cell_information.normal_vector);
+              auto point_on_boundary =
+                boundary_cell_information.point_on_boundary;
 
-          // Finding the projected vector on the normal vector of the boundary.
-          // Here we have used the private function find_projection. Using this
-          // projected vector, the particle-wall distance is calculated
-          Tensor<1, 3> projected_vector =
-            particle_wall_contact_force_object->find_projection(
-              point_to_particle_vector, normal);
+              // Calculation of gap
+              Point<3> particle_position_3d = point_nd_to_3d(particle.position);
+              Point<3> point_on_boundary_3d = point_nd_to_3d(point_on_boundary);
 
+              Tensor<1, 3> point_to_particle_vector =
+                particle_position_3d - point_on_boundary_3d;
 
-          double gap = (projected_vector.norm()) - particle.radius;
+              // Finding the projected vector on the normal vector of the
+              // boundary. Here we have used the private function
+              // find_projection. Using this projected vector, the
+              // particle-wall distance is calculated
+              Tensor<1, 3> projected_vector =
+                ((point_to_particle_vector * normal) / (normal.norm_square())) *
+                normal;
 
-          Tensor<1, 3> radial_vector;
-          double       radial_velocity;
-          Tensor<1, 3> f_lub;
+              double gap = (projected_vector.norm()) - particle.radius;
 
+              Tensor<1, 3> radial_vector;
+              double       radial_velocity;
+              Tensor<1, 3> f_lub;
 
-          radial_vector = projected_vector;
-          if (gap > 0 and gap < h_max)
-            // This means that the particles is very close to the wall but not
-            // in contact
-            {
-              // Limit the smallest gap calculated
-              if (gap < h_min)
+              radial_vector = projected_vector;
+              if (gap > 0 and gap < h_max)
+                // This means that the particles is very close to the wall
+                // but not in contact
                 {
-                  gap = h_min;
+                  // Limit the smallest gap calculated
+                  if (gap < h_min)
+                    {
+                      gap = h_min;
+                    }
+                  // Evaluate the force
+                  radial_velocity =
+                    scalar_product(-radial_vector, particle.velocity);
```

Comment:
shouldn't the radial_vector be divided by its norm here ? You do it three lines below (948) to project the lubrication force on the radial axis. So if you just want the radial component of the velocity there is a / radial_vector.norm() missing